carbonmark/components/pages/Retire/RetireModal/index.tsx

```diff
@@ -0,0 +1,146 @@
+
+  const onModalClose = !isPending ? props.onCloseModal : undefined;
+
+  const RetireApproval = () => {
+    return (
+      <div className={styles.formatParagraph}>
+        <Text t="caption">
+          <Trans id="transaction_modal.approve.allow_amount_1">
+            You are about to retire a carbon asset.
+          </Trans>
+        </Text>
+        <Text t="caption">
+          <Trans id="transaction_modal.approve.allow_amount_2">
+            The first step is to grant the approval to transfer your carbon
+            asset from your wallet to Carbonmark, the next step is to approve
+            the actual transfer and complete your retirement.
+          </Trans>
+        </Text>
+        <Text t="caption">
+          <Trans id="transaction_modal.approve.allow_amount_3">
+            Verify all information is correct and click 'approve' to continue.{" "}
+          </Trans>
+        </Text>
+      </div>
+    );
+  };
+
+  const RetireSubmit = () => {
+    return (
+      <div className={styles.formatParagraph}>
+        <Text t="caption">
+          <Trans id="transaction_modal.submit.confirm_transaction_1">
+            The previous step granted the approval to transfer your carbon asset
+            from your wallet to Carbonmark, your retirement has not been
+            completed yet.
+          </Trans>
+        </Text>
+        <Text t="caption">
+          <Trans id="transaction_modal.submit.confirm_transaction_2">
+            To finalize your retirement, verify all information is correct and
+            then click 'submit' below.
+          </Trans>
+        </Text>
+      </div>
+    );
+  };
+
+  return (
+    <CustomizableModal
+      title={
+        processingRetirement ? (
+          <div>
+            <Text className={styles.processingTitle} t="h3">
+              Processing Retirement
+            </Text>
+          </div>
+        ) : (
+          props.title
+        )
+      }
+      maxWidth={processingRetirement ? "43rem" : "50rem"}
+      height={processingRetirement ? "26rem" : "fit-content"}
+      maxHeight="calc(100vh - 8rem)"
+      showModal={props.showModal}
+      onToggleModal={processingRetirement ? undefined : onModalClose}
+    >
+      {processingRetirement ? (
+        <div className={styles.processingRetirement}>
+          <Spinner className={styles.processingSpinner} />
+        </div>
+      ) : (
+        <Transaction
+          hasApproval={props.isApproved}
+          amount={{
+            value: props.value,
+            token: props.token.key as CarbonmarkToken,
+          }}
+          spenderAddress={props.spenderAddress}
+          onApproval={props.onApproval}
+          onSubmit={props.onSubmit}
+          onCancel={props.onCloseModal}
+          status={props.status}
+          onResetStatus={props.onResetStatus}
+          approvalText={<RetireApproval />}
+          submitText={<RetireSubmit />}
+          onGoBack={() => {
```

Comment:
Here you do not need a Go Back button.
The user can simply close the modal and see the inputs again.

The Go Back button is needed when the input form was in the same modal before.
But this is not the case here.

Please add the Figma designs again to the PR description if I am missing something here.

Reply:
this definitely makes sense but I had included the Go Back button as it was included in the design (below):

<img width="252" alt="Screen Shot 2023-05-30 at 10 55 15 AM" src="https://github.com/KlimaDAO/klimadao/assets/72105234/87203ee7-db9a-4660-a8e8-4009bc2536cc">

I agree there's not really a need for it however